packages/engine-server/src/engineClient.ts

```diff
@@ -322,6 +322,12 @@ export class DendronEngineClient implements DEngineClient {
     });
   }
 
+  /** Renames the note.
+   *
+   *  WARNING: When doing bulk operations. Do not invoke multiple requests to this
+   *  command in parallel, wait for a single call to finish before requesting another call.
+   *  Otherwise some race condition starts to cause intermittent failures.
```

Comment:
Is this something that will affect end users? If so, it should be noted in the docs when we ship it. or ideally, prevent it from happening internally.

Reply:
As long as we call this in sequence it should work as expected. I was running into intermittent issue when trying to issue multiple calls at the same time and then wait for all of their completion

Waiting for a single change to fully complete (as this method is currently used in this PR) should work all the time without this intermittent issue:

```
 for (const move of necessaryMoves) {
      // We need to wait for a rename to finish before triggering another rename
      // eslint-disable-next-line no-await-in-loop
      const changes = await engine.renameNote(move);

      allChanges.push(...(changes.data as NoteChangeEntry[]));
    }
```

The warning is added so we ourselves use this renameNote() only in sequence and not in parallel.

Reply:
got it!